sktime/decomposition/koopman.py

```diff
@@ -0,0 +1,575 @@
+            x = self.basis_transform_forward(trajectory[t])  # map into basis in which K is defined
+            x = operator.T @ x  # apply operator, we are now in the modified basis of g
+            x = self.basis_transform_backward(x, inverse=True)  # map back to observable basis
+            output_trajectory[t] = x
+        return output_trajectory
+
+    def transform(self, data, forward=True, propagate=False, **kwargs):
+        r"""Projects the data into the Koopman operator basis, possibly discarding non-relevant dimensions.
+
+        Beware that only `frames[tau:]` of each trajectory returned by this method contain valid values
+        of the "future" functions :math:`g(x_{t+\tau})`. Conversely, only `frames[:-tau]` of each
+        trajectory returned by this method contain valid values of the "instantaneous" functions :math:`f(x_t)`.
+        The remaining frames could possibly be interpreted as some extrapolation.
+
+        Parameters
+        ----------
+        data : ndarray(T, n)
+            the input data
+        forward : bool, default=True
+            Whether to use forward or backward transform for projection.
+        propagate : bool, default=False
+            Whether to propagate the projection with :math:`K^\top` (or :math:`(K^\top)^{-1}` in the backward case).
+
+        Returns
+        -------
+        projection : ndarray(T, m)
+            The projected data.
+            In case of VAMP, if `forward` is True, projection will be on the right singular
+            functions. Otherwise, projection will be on the left singular functions.
+        """
+        if forward:
+            transform = self.basis_transform_forward(data, dim=self.output_dimension)
+            return self.operator.T @ transform if propagate else transform
+        else:
+            transform = self.basis_transform_backward(data, dim=self.output_dimension)
+            return self.operator_inverse.T @ transform if propagate else transform
+
+
+class CovarianceKoopmanModel(KoopmanModel):
+    r"""A type of Koopman model which was obtained through diagonalization of covariance matrices. This leads to
+    a Koopman operator which is a diagonal matrix and can be used to project onto specific processes of the system.
+
+    The estimators which produce this kind of model are :class:`VAMP <sktime.decomposition.VAMP>` and
+    :class:`TICA <sktime.decomposition.TICA>`."""
+
+    def __init__(self, operator: np.ndarray, basis_transform_forward: Optional[KoopmanBasisTransform],
+                 basis_transform_backward: Optional[KoopmanBasisTransform], cov: CovarianceModel,
+                 rank_0: int, rank_t: int, dim=None, scaling=None, epsilon=1e-6):
+        r""" For a description of parameters `operator`, `basis_transform_forward`, `basis_transform_backward`,
+        and `output_dimension`: please see :meth:`KoopmanModel.__init__`.
+
+        Parameters
+        ----------
+        cov : CovarianceModel
```

Comment:
`cov` is a model?

Reply:
yes on both accounts, cov is a model containing `cov_00`, `cov_0t`, `cov_tt` and instantaneous as well as time-lagged means